include/deal.II/base/function_parser.h

```diff
@@ -397,18 +397,39 @@ class FunctionParser : public AutoDerivativeFunction<dim>
 
 private:
 #ifdef DEAL_II_WITH_MUPARSER
+
   /**
-   * Place for the variables for each thread
+   * std::is_copy_constructible gives the wrong answer for containers with
+   * non-copy constructible types (e.g., std::vector<std::unique_ptr<int>>).
+   * Work around that problem by defining our own explicitly uncopyable
+   * per-thread data class (instead of, e.g., using std::pair).
+   *
+   * We define exactly one per-thread data structure to avoid locking and
+   * unlocking the TLS data lookup more than once per function call.
```

Comment:
I find this comment difficult to read because it starts the class documentation by presenting a problem that does not actually exist in our code. In other words, it documents something that this class explicitly does not implement.

It would be easier to read if you followed the order
* what does this class do
* why is it designed this way
* why is it *not* designed that other way

Reply:
You don't like my stream-of-consciousness documentation? I'll fix it :)

Reply:
Fixed!